cpmpy/solvers/choco.py

```diff
@@ -535,6 +536,15 @@ def _get_constraint(self, cpm_expr):
                 # Create task variables. Choco can create them only one by one
                 tasks = [self.chc_model.task(s, d, e) for s, d, e in zip(start, dur, end)]
                 return self.chc_model.cumulative(tasks, demand, cap)
+            elif cpm_expr.name == "subcircuit":
+                # Successor variables
+                succ = self.solver_vars(cpm_expr.args)
+                # Start index must be in subcircuit and thus cannot selfloop
+                if cpm_expr.startIndex is not None:
+                    self += [(cpm_expr.args[cpm_expr.startIndex] != cpm_expr.startIndex)]
+                # Add an unused variable for the subcircuit length, assume that subcircuit should consist of at least 2 nodes.
+                subcircuit_length = self.chc_model.intvar(2, len(succ))  
```

Comment:
Don't make integer variables on the Choco solver directly, create a CPMpy intvar and go through `self.solver_var` instead